tests/unit/acquisition/test_rule.py

```diff
@@ -280,7 +283,7 @@ def test_trust_region_for_unsuccessful_global_to_local_trust_region_unchanged()
 
 
 def test_trust_region_for_successful_local_to_global_trust_region_increased() -> None:
-    tr = TrustRegion(NegativeLowerConfidenceBound(0).using(OBJECTIVE))
+    tr = TrustRegion(EfficientGlobalOptimization(NegativeLowerConfidenceBound(0)))
```

Comment:
Maybe add some tests for `TrustRegion(DiscreteThompsonSampling(...))` as that would probe the usefulness of this change?

Reply:
hmmm ... fair point. I'll have to think how much I can be bothered. That's quite a non-trivial thing to do

Reply:
i've used a stub acquisition rule so I don't need to understand thompson sampling